x-pack/plugins/security_solution/public/detections/components/take_action_dropdown/index.tsx

```diff
@@ -183,69 +181,91 @@ export const TakeActionDropdown = React.memo(
       [eventFilterActions, exceptionActions, isEvent, actionsData.ruleId]
     );
 
-    const addToCaseProps = useMemo(() => {
-      if (ecsData) {
-        return {
-          event: { data: [], ecs: ecsData, _id: ecsData._id },
-          useInsertTimeline: insertTimelineHook,
-          casePermissions,
-          appId: APP_ID,
-          onClose: afterCaseSelection,
-        };
-      } else {
-        return null;
-      }
-    }, [afterCaseSelection, casePermissions, ecsData, insertTimelineHook]);
+    const addToCaseProps = useMemo(
+      () => ({
+        event: { data: detailsData, ecs: ecsData, _id: ecsData._id },
```

Comment:
it throws an error for `_id: ecsData._id` on case details page as ecsData is not available there. so I think we couldn't assume ecsData is always available here.